### PR TITLE
Fix PhpDocs (Noticed with PhpStan Level 5)

### PR DESCRIPTION
**Pull request by blankse**
Fix only PhpDocs for branch 10.3.
Splitting up from #11826

Fix 129 PhpStan level 5 errors.

### CODE REVIEW

File: bundles/EcommerceFrameworkBundle/CartManager/CartInterface.php [hunk 1]
```diff
@@ -40,12 +40,12 @@ interface CartInterface
     const COUNT_MAIN_AND_SUB_ITEMS = 'main_and_sub';
 
     /**
-     * @return int|null
+     * @return string|int|null
```
Comment (jdreesen) [inline bundles/EcommerceFrameworkBundle/CartManager/CartInterface.php:43]: Something is not right throughout the files. Sometimes it's only `string`, sometimes it's only `int`, sometimes both.

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData.php [hunk 2]
```diff
@@ -51,6 +51,9 @@ public static function getByKeyCartId($key, $cartId)
         return $checkoutDataItem;
     }
 
+    /**
+     * @param int $cartId
```
Comment (jdreesen) [inline bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData.php]: Here it's only `int`

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Dao.php [hunk 3]
```diff
@@ -106,6 +106,9 @@ public function delete()
         $this->db->deleteWhere(self::TABLE_NAME, '`key`=' . $this->db->quote($this->model->getKey()). ' AND cartId = ' . $this->db->quote($this->model->getCartId()));
     }
 
+    /**
+     * @param int $cartId
```
Comment (jdreesen) [inline bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Dao.php]: Here it's only `int`, too

File: bundles/EcommerceFrameworkBundle/CartManager/CartItemInterface.php [hunk 4]
```diff
@@ -130,7 +130,7 @@ public static function getByCartIdItemKey($cartId, $itemKey, $parentKey = '');
     /**
      * @static
      *
-     * @param int $cartId
+     * @param string $cartId
```
Comment (jdreesen) [inline bundles/EcommerceFrameworkBundle/CartManager/CartItemInterface.php]: Here it's only `string`

File: bundles/EcommerceFrameworkBundle/CartManager/CartInterface.php (hunk 1 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/CartManager/CartInterface.php:43]: SessionCart use string as id see:
https://github.com/pimcore/pimcore/blob/87972b48f4293cd1b54641ef3c2fc708a63de718/bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php#L61
Cart use int as id because the ecommerceframework_cart db table id column is an integer:
https://github.com/pimcore/pimcore/blob/87972b48f4293cd1b54641ef3c2fc708a63de718/bundles/EcommerceFrameworkBundle/CartManager/Cart/Dao.php#L69

So the interface should allow both at the moment.

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Dao.php (hunk 3 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Dao.php]: Is only used for the Cart. So int should be enough

Comment (blankse) [reply bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Dao.php]: I set it everywhere to string|int now: https://github.com/pimcore/pimcore/pull/11828/commits/6a1132b3daf330b26023f8d62b431ca60e19d144

File: bundles/EcommerceFrameworkBundle/CartManager/CartItemInterface.php (hunk 4 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/CartManager/CartItemInterface.php]: I set it everywhere to string|int now: https://github.com/pimcore/pimcore/pull/11828/commits/6a1132b3daf330b26023f8d62b431ca60e19d144

File: bundles/AdminBundle/Controller/Admin/Document/SnippetController.php [hunk 5]
```diff
@@ -146,6 +146,7 @@ public function saveAction(Request $request)
         if ($snippetSession) {
             $snippet = $snippetSession;
         } else {
+            /** @var Document\Snippet $snippet */
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/Document/SnippetController.php]: can we just replace line 143? I don't think so we need to define `$snippetSession` after this change

File: bundles/EcommerceFrameworkBundle/CartManager/AbstractCartItem.php [hunk 6]
```diff
@@ -144,15 +144,15 @@ public function setCart(CartInterface $cart)
     abstract public function getCart();
 
     /**
-     * @return int
+     * @return string|int
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/CartManager/AbstractCartItem.php]: It can be null if just do:

```
$cartItemName = $cart->getCartItemClassName():
$cartItem = new $cartItemName();
$cartItem->getCartId(); //returns null
```

File: bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php [hunk 7]
```diff
@@ -55,7 +55,7 @@ abstract public function getOrderByDesc(): ?string;
      *
      * @abstract
      *
-     * @return \Pimcore\Model\DataObject\Fieldcollection|null
+     * @return AbstractFilterDefinitionType[]|null
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php]: why this change? `AbstractFilterDefinitionType[]` is items property in the returned object

image

File: bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php [hunk 8]
```diff
@@ -64,7 +64,7 @@ abstract public function getConditions();
...
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php]: see above comment

File: bundles/EcommerceFrameworkBundle/PricingManager/Rule.php [hunk 9]
```diff
@@ -60,7 +60,7 @@ public static function getById($id)
     }
 
     /**
-     * @var int
+     * @var int|null
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/PricingManager/Rule.php:63]: Should be changed in RuleInterface as well?

File: models/Asset/Document/ImageThumbnail.php [hunk 10]
```diff
@@ -40,7 +43,7 @@ final class ImageThumbnail
     protected $page = 1;
 
     /**
-     * @param Model\Asset\Document $asset
+     * @param Model\Asset\Document|null $asset
```
Comment (dvesh3) [inline models/Asset/Document/ImageThumbnail.php:46]: why asset is nullable here?

Comment (dvesh3) [inline models/Asset/Document/ImageThumbnail.php:46]: why asset is nullable here?

File: models/Asset/Video/ImageThumbnail.php [hunk 11]
```diff
@@ -25,6 +25,9 @@
 use Symfony\Component\EventDispatcher\GenericEvent;
 use Symfony\Component\Lock\LockFactory;
 
+/**
+ * @property Model\Asset\Video|null $asset
```
Comment (dvesh3) [inline models/Asset/Video/ImageThumbnail.php:29]: why asset is nullable here?

File: models/DataObject/AbstractObject.php [hunk 12]
```diff
@@ -1062,7 +1062,7 @@ public function getUserModification()
     }
 
     /**
-     * @param int $o_id
+     * @param int|null $o_id
```
Comment (dvesh3) [inline models/DataObject/AbstractObject.php:1065]: why null here?

File: bundles/EcommerceFrameworkBundle/CartManager/AbstractCartItem.php (hunk 6 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/CartManager/AbstractCartItem.php]: Changed it https://github.com/pimcore/pimcore/pull/11828/commits/9db93bee2fc05661884f954c1cb56334ae29d9c1

File: bundles/EcommerceFrameworkBundle/PricingManager/Rule.php (hunk 9 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/PricingManager/Rule.php:63]: Changed also RuleInterface https://github.com/pimcore/pimcore/pull/11828/commits/cc559db55de56506f6d3ad5f1c70e186ed716e31

File: models/Asset/Document/ImageThumbnail.php (hunk 10 above)
Comment (blankse) [reply models/Asset/Document/ImageThumbnail.php:46]: See for example https://github.com/pimcore/pimcore/blob/89ecd1b81d4e955164027ca3a9129fa4b0d82679/models/Asset/Document.php#L97

File: models/Asset/Video/ImageThumbnail.php (hunk 11 above)
Comment (blankse) [reply models/Asset/Video/ImageThumbnail.php:29]: See https://github.com/pimcore/pimcore/blob/89ecd1b81d4e955164027ca3a9129fa4b0d82679/models/Asset/Video.php#L171

File: models/DataObject/AbstractObject.php (hunk 12 above)
Comment (blankse) [reply models/DataObject/AbstractObject.php:1065]: See https://github.com/pimcore/pimcore/blob/89ecd1b81d4e955164027ca3a9129fa4b0d82679/models/DataObject/Service.php#L215

In https://github.com/pimcore/pimcore/pull/11826 I changed also the method body, so it will not cast to 0 on copy

File: bundles/AdminBundle/Controller/Admin/Document/SnippetController.php (hunk 5 above)
Comment (blankse) [reply bundles/AdminBundle/Controller/Admin/Document/SnippetController.php]: I used now PHPDoc Generics for getLastVersion, so PhpStorm/PhpStan knows, the param type is the same as the return type: https://github.com/pimcore/pimcore/pull/11828/commits/bcebe6b44d9a17af02218bc45a5a4d07a0e55a20

File: bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php (hunk 7 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php]: You are right. I use here class level generic now. So we can set the item type in the declaration:
https://github.com/pimcore/pimcore/pull/11828/commits/5da24d8c38b990b7669ecb0a4d7cfc859bdb5d3d
https://github.com/pimcore/pimcore/pull/11828/commits/72b9fc22dd809f0fb4ebefb6ba6072c04aba1443

File: bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php (hunk 8 above)
Comment (blankse) [reply bundles/EcommerceFrameworkBundle/Model/AbstractFilterDefinition.php]: See above comment :)

**Comment by blankse** [comment]
@dvesh3 Thanks for review. I commented/changed all suggestions. Can you look again?

**Comment by dvesh3** [comment]
> @dvesh3 Thanks for review. I commented/changed all suggestions. Can you look again?

LGTM. thanks!

**Review by dvesh3 — APPROVED** [review]
